src/client/datascience/progress/kernelProgressReporter.ts

```diff
@@ -127,6 +122,16 @@ export class KernelProgressReporter implements IExtensionSyncActivationService {
             progressInfo.progressList.push(title);
             progressInfo.reporter.report({ message: title });
         } else {
+            // if we've display this message in the past, then no need to display it again.
+            // Due to the async nature of things, we may have already displayed it and we don't want to display it again.
+            // Also displaying the same message again & again could confuse the user.
+            // It could look as though the same operation is being performed multiple times (when in fact its possible we have caching in place).
+            // Eg. we could be attempting to start a python process, which requires activation, thats cached, however calling it multiple times
+            // could result in multiple messages being displayed.
+            // Perhaps its the right thing to do and display the message multiple times, but for now, we'll just not display it.
```

Comment:
We can change this if required, felt weird to display the messages a number of times.
E.g. we have messages displayed as follows:
* Activating Python environment abc
* Validating kernel dependencies
* Activating Python environment  abc (again)

This change ensures we display it just once.